src/Build/Evaluation/Expander.cs

```diff
@@ -3416,7 +3484,26 @@ private bool TryExecuteWellKnownFunction(out object returnVal, object objectInst
                 }
                 else if (objectInstance == null)
                 {
-                    if (_receiverType == typeof(Math))
+                    if (_receiverType == typeof(string))
+                    {
+                        if (string.Equals(_methodMethodName, nameof(string.IsNullOrWhiteSpace), StringComparison.OrdinalIgnoreCase))
+                        {
+                            if (TryGetArg(args, out string arg0))
+                            {
+                                returnVal = string.IsNullOrWhiteSpace(arg0);
+                                return true;
+                            }
+                        }
+                        else if (string.Equals(_methodMethodName, nameof(string.IsNullOrEmpty), StringComparison.OrdinalIgnoreCase))
```

Comment:
Do you recall where this and `IsNullOrWhiteSpace` got called? I wonder if we could use `[MSBuild]::ValueOrDefault` to simplify the invocations.

Reply:
Sadly the Expander.Function object does not know the IElementLocation. But for experimentation's sake I'll pipe the location down and log it.

Reply:
Could also just grep through a preprocessed project or two; I don't feel super strongly about this.

Reply:
Apparently it wasn't very hard to tweak the code to print the location. Sadly the callsites cannot be converted to `[MSBuild]::ValueOrDefault`:

https://github.com/Microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L75

https://github.com/Microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L5998